onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -943,6 +943,18 @@ class PlannerImpl {
               }
             }
           }
+#if defined(ENABLE_TRAINING) && defined(ENABLE_TRAINING_ON_DEVICE)
+          if (pnode->OpType() == "InPlaceAccumulator") {
+            const NodeArg* input = pnode->InputDefs()[0];
+            const auto& input_name = input->Name();
+            const auto input_index = Index(input_name);
+
+            const auto& alloc_plan = AllocPlan(input_index);
+            if (alloc_plan.alloc_kind == AllocKind::kPreExisting) {
+              Reuse(input_index, current, AllocKind::kShare);
```

Comment:
could you pls elaborate about what issues this targets to fix?

Reply:
This is the link for topic we discussed around in the meeting. FYI. https://github.com/microsoft/onnxruntime/pull/11506#discussion_r875796187

Reply:
adding some comments around this area. We can change this if we change the InPlaceAccumulator op with optional outputs.